File: backend/src/main/java/outcobra/server/model/Holiday.java
package outcobra.server.model;

import java.time.LocalDate;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import lombok.*;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class Holiday {
    @Id
    private Long id;

    @NotNull
    private String name;

    @NotNull
    private LocalDate validFrom, validTo;

    @ManyToOne
    private SchoolYear schoolYear;
}

File: backend/src/main/kotlin/outcobra/server/model/domain/Holiday.kt
package outcobra.server.model.domain

import outcobra.server.model.interfaces.ParentLinked
import java.time.LocalDate
import javax.persistence.Entity
import javax.persistence.ManyToOne
import javax.validation.constraints.NotNull

@Entity
data class Holiday(@NotNull var name: String? = null,
                   @NotNull var validFrom: LocalDate = LocalDate.now(),
                   @NotNull var validTo: LocalDate = LocalDate.now(),
                   @ManyToOne var schoolYear: SchoolYear? = null)
    : ParentLinked, AbstractEntity() {

    override val parent: ParentLinked?
        get() = schoolYear
}